File: DictionaryProject/WordBuilder/Project.vb
Public Class Project

    Private _Rules As New RuleCollection()
    Private _TokenSets As New TokenSetCollection()
    Public ReadOnly Property TokenSets() As TokenSetCollection
        Get
            Return _TokenSets
        End Get
    End Property

    Public ReadOnly Property Rules() As RuleCollection
        Get
            Return _Rules
        End Get
    End Property

    Private _columns As New Dictionary(Of String, String)
    Public ReadOnly Property Columns() As Dictionary(Of String, String)
        Get
            Return _columns
        End Get
    End Property

    Private _startRules As New Dictionary(Of String, Integer)
    Public ReadOnly Property StartRules() As Dictionary(Of String, Integer)
        Get
            Return _startRules
        End Get
    End Property

    Public Function GetWord(ByVal startRule As String) As Context
        Dim c As New Context()

        Dim r As Rule = _Rules.GetRuleByName(startRule)

        If r IsNot Nothing Then
            r.Execute(c)
        End If

        Return c
    End Function

    Private _Warnings As New List(Of String)
    Public ReadOnly Property Warnings() As List(Of String)
        Get
            Return _Warnings
        End Get
    End Property
End Class

File: monowordbuilder/WordBuilderProject/Project.cs
using System;
using System.Collections.Generic;

public class Project
{	
	private RuleCollection _Rules = new RuleCollection();
	
	private TokenSetCollection _TokenSets = new TokenSetCollection();

	public TokenSetCollection TokenSets {
		get { return _TokenSets; }
	}
	
	public RuleCollection Rules {
		get { return _Rules; }
	}
	
	private Dictionary<string, string> _columns = new Dictionary<string, string>();
	public Dictionary<string, string> Columns {
		get { return _columns; }
	}
	
	private Dictionary<string, int> _startRules = new Dictionary<string, int>();
	public Dictionary<string, int> StartRules {
		get { return _startRules; }
	}
	
	public Context GetWord(string startRule)
	{
		Context c = new Context();
		
		try {
			Rule r = _Rules.GetRuleByName(startRule);
			
			if (r != null) {
				r.Execute(c);
			}
		}
		catch (Exception ex) {
			Warnings.Add(ex.Message);
		}
		
		return c;
	}
	
	private List<string> _Warnings = new List<string>();
	public List<string> Warnings {
		get { return _Warnings; }
	}
}
